tests/run-make/mixing-libs/rmake.rs

```diff
@@ -0,0 +1,21 @@
+// Having multiple upstream crates available in different formats
+// should result in failed compilation. This test causes multiple
+// libraries to exist simultaneously as rust libs and dynamic libs,
+// causing prog.rs to fail compilation.
+// See https://github.com/rust-lang/rust/issues/10434
+
+//@ ignore-cross-compile
+
+use run_make_support::{remove_dylibs, rustc};
+
+fn main() {
+    rustc().input("rlib.rs").crate_type("rlib").crate_type("dylib").run();
+
+    // Not putting -C prefer dynamic here allows for static linking of librlib.rlib.
```

Comment:
I believe this should probably be 
```suggestion
    // Not putting `-C prefer-dynamic` here allows for static linking of librlib.rlib.
```
because I had to look back at the makefile to determine if it was prefer=dynamic (I didn't think it was)

Reply:
Good catch! Thanks.